src/measure_theory/integration.lean

```diff
@@ -650,102 +727,91 @@ begin
     exact with_top.zero_lt_top }
 end
 
-/-- A technical lemma dealing with the definition of `integrable` in `l1_space.lean`. -/
-lemma integral_map_coe_lt_top {f : α →ₛ β} {g : β → nnreal} (h : f.fin_vol_supp) (hg : g 0 = 0) :
-  integral (f.map ((coe : nnreal → ennreal) ∘ g)) < ⊤ :=
-integral_lt_top_of_fin_vol_supp
-  (by { filter_upwards[], assume a, simp only [mem_set_of_eq, map_apply], exact ennreal.coe_lt_top})
-  (by { apply fin_vol_supp_map h, simp only [hg, function.comp_app, ennreal.coe_zero] })
+end fin_meas_supp
 
-end fin_vol_supp
+end fin_meas_supp
 
 end simple_func
 
 section lintegral
 open simple_func
-variable [measure_space α]
-
-/-- The lower Lebesgue integral -/
-def lintegral (f : α → ennreal) : ennreal :=
-⨆ (s : α →ₛ ennreal) (hf : ⇑s ≤ f), s.integral
+variables [measurable_space α] {μ : measure α}
 
-notation `∫⁻` binders `, ` r:(scoped f, lintegral f) := r
+/-- The lower Lebesgue integral of a function `f` with respect to a measure `μ`. -/
+def lintegral (μ : measure α) (f : α → ennreal) : ennreal :=
+⨆ (g : α →ₛ ennreal) (hf : ⇑g ≤ f), g.lintegral μ
 
-theorem simple_func.lintegral_eq_integral (f : α →ₛ ennreal) : (∫⁻ a, f a) = f.integral :=
-le_antisymm
-  (supr_le $ assume s, supr_le $ assume hs, integral_le_integral _ _ hs)
-  (le_supr_of_le f $ le_supr_of_le (le_refl f) $ le_refl _)
+notation `∫⁻` binders ` in ` s `, ` r:(scoped:60 f, f) ` ∂` μ:70 :=
+  lintegral (measure.restrict s μ) r
+notation `∫⁻` binders ` in ` s `, ` r:(scoped:60 f, lintegral (measure.restrict s volume) f) := r
```

Comment:
The problem with pretty-printing this notation is that there is a hidden coercion here:
```
∫⁻ (x : α), 42 ∂⇑(measure.restrict A) μ : ennreal
```
The pretty-printer won't find this notation because it does not match (there is no `⇑` in the notation).  Putting the `⇑` in the notation doesn't work because then the elaborator fails (but that could be fixed).

This would work:
```
notation `∫⁻` binders `, ` r:(scoped:60 f, f) ` ∂` μ:70 := lintegral μ r
notation `∫⁻` binders `, ` r:(scoped:60 f, lintegral volume f) := r

notation `∫⁻` binders ` in ` s `, ` r:(scoped:60 f,
  lintegral (linear_map.to_fun (measure.restrict s) volume) f) := r
notation `∫⁻` binders ` in ` s `, ` r:(scoped:60 f, f) ` ∂` μ:70 :=
  lintegral (linear_map.to_fun (measure.restrict s) μ) r
```

Reply:
I'll just make `measure.restrict` a function and `measure.restrictₗ` the linear map.

Reply:
@urkud If you want the linear map to be the "default" version to use, what about `measure.restrict_fun` (function) and `measure.restrict` (linear map)...